dftimewolf/lib/exporters/gce_disk_export.py

```diff
@@ -72,32 +81,64 @@ def SetUp(self,
+    If remote_instance_name is specified, two behaviors are possible:
+    * If no other parameters are specified, it will select the instance's boot
+      disk
+    * if all_disks is set to True, it will select all disks in the project
+      that are attached to the instance
+
     Args:
       source_project_name (str): Source project ID containing
           the disk to export.
-      source_disk_name (str): Disk that needs to be exported.
       gcs_output_location (str): Google Cloud Storage parent bucket/folder
           path of the exported image.
-      analysis_project_name (Optional[str]): Project ID where the
-          disk image is created then exported. If not exit,
+      analysis_project_name (Optional[str]): Optional. Project ID where the
+          disk image is created then exported. Default,
           source_project_name will be used.
-      exported_disk_name (Optional[str]): Name of the output file, must comply
-          with ^[A-Za-z0-9-]*$' and '.tar.gz' will be appended to the name.
-          If not exist, random name will be generated.
+      disk_names (Optional[str]): Optional. Comma separated disk names to copy.
+          Default is None.
+      remote_instance_name (Optional[str]): Optional. Instance in source
+          project to export its disks. Default, if not exist, disk_names
+          will be used.
+      all_disks (Optional[bool]): Optional. True if all disks attached to
+          the source instance should be copied. Default is False. If False
+          and remote_instance_name is provided it will select the instance's
+          boot disk.
+      exported_image_name (Optional[str]): Optional. Name of the output file,
+          must comply with ^[A-Za-z0-9-]*$' and '.tar.gz' will be appended to
+          the name. Default is None, if not exist or if more than one disk
+          is selected, exported image name as
+          "exported-image-{TIMESTAMP('%Y%m%d%H%M%S')}".
     """
     self.source_project = gcp_project.GoogleCloudProject(source_project_name)
     if analysis_project_name:
       self.analysis_project = gcp_project.GoogleCloudProject(
           analysis_project_name)
     else:
       self.analysis_project = self.source_project
-    self.source_disk = self.source_project.compute.GetDisk(source_disk_name)
+
+    if not (remote_instance_name or disk_names):
+      self.state.AddError(
+          'You need to specify at least an instance name or disks to copy',
+          critical=True)
+      return
+
+    init_state = state_mod.DFTimewolfState(config.Config)
+    gcloud_collector = gcloud.GoogleCloudCollector(init_state)
+    gcloud_collector.SetUp(
+        analysis_project_name=self.analysis_project.project_id,
+        remote_project_name=self.source_project.project_id,
+        create_analysis_vm=False,
+        disk_names=disk_names,
+        all_disks=all_disks,
+        remote_instance_name=remote_instance_name)
+
+    self.source_disks = gcloud_collector.FindDisksToCopy()
```

Comment:
I agree that it is better to avoid doing things this way, I just wanted to avoid copying FindDisksToCopy since I will need to copy _GetDisksFromInstance and _GetDisksFromNames as well, but I will copy them and do it in a more proper way.